Source/Core/InputCommon/DynamicInputTextureConfiguration.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <map>
+#include <string>
+
+#include "Common/CommonTypes.h"
+#include "Common/IniFile.h"
+
+namespace InputCommon
+{
+class DynamicInputTextureConfiguration
+{
+public:
+  explicit DynamicInputTextureConfiguration(const std::string& json_file);
+  ~DynamicInputTextureConfiguration();
+  void GenerateTextures(const IniFile::Section* sec);
+
+private:
+  struct DynamicInputTextureData
```

Comment:
I like this idea but the vector's type but VisualStudio complains about this change.  I did a similar forward declaration of `DynamicInputTextureConfiguration` in `DynamicInputTextureManager`'s header but with this forward declaration it complains there about not having the size of `DynamicInputTextureConfiguration`.  I'm not too familiar with this style of forward declaration, I mostly forward declare pointer types, so I might be missing something.

Reply:
Can you provide both the header and source file with the changes you made that caused the compilation error?

Reply:
@lioncash - please see the last commit on #8318

Reply:
To give an example:

```
// a.h
struct B;
struct A
{
 A();
 ~A();
 B b;
};
void do_something(const& A a);

// a.cpp
#include "b.h"
A::A() = default;
A::~A() = default;
void do_something(const& A a)
{
  // ...
}

// b.h
struct B
{
  B();
  ~B();
  struct Imp;
  imp imp;
};

// B.cpp
struct B::Imp
{
  int a = 3;
};
B::B() = default;
B::~B() = default;
```

(didn't actually compile this but it portrays the basic idea) We can't compile "A" because "B" entire size isn't known from within A's cpp file

Reply:
I reverted the change (since it wasn't building as is), it only saves a map include.  Let me know though if I'm missing something.